### PR TITLE
[SCISPARK- 63][ESIP workshop]include setMask function to change default mask value 

### PR DESCRIPTION
**Pull request by rahulpalamuttam**
Issue #63 
add setMask function to change default mask value on sciTensor and Abstract tensors

### CODE REVIEW

File: src/main/scala/org/dia/core/SciTensor.scala [hunk 1]
```diff
@@ -119,37 +119,43 @@ class SciTensor(val variables: mutable.HashMap[String, AbstractTensor]) extends
...
 
-/**
+  /**
    * Masks the current variable array by preserving values
    * less than to num.
    */
   def <(num: Double): SciTensor = variables(varInUse) < num
 
-/**
+  /**
```
Comment (chrismattmann) [inline src/main/scala/org/dia/core/SciTensor.scala]: orthogonal change please remove

File: src/main/scala/org/dia/core/SciTensor.scala [hunk 2]
```diff
@@ -119,37 +119,43 @@ class SciTensor(val variables: mutable.HashMap[String, AbstractTensor]) extends
...
    * Masks the current variable array by preserving values
    * greater than or equal to num.
    */
   def >=(num: Double): SciTensor = variables(varInUse) >= num
 
-/**
+  /**
```
Comment (chrismattmann) [inline src/main/scala/org/dia/core/SciTensor.scala]: orthogonal change please remove

File: src/main/scala/org/dia/core/SciTensor.scala [hunk 3]
```diff
@@ -119,37 +119,43 @@ class SciTensor(val variables: mutable.HashMap[String, AbstractTensor]) extends
...
+  def setMask(num: Double): SciTensor = variables(varInUse).setMask(num)
+
   /**
    * Masks the current variable array by preserving values
    * less than or equal to num.
    */
   def <=(num: Double): SciTensor = variables(varInUse) <= num
 
-/**
+  /**
```
Comment (chrismattmann) [inline src/main/scala/org/dia/core/SciTensor.scala]: orthogonal change please remove

File: src/main/scala/org/dia/core/SciTensor.scala [hunk 4]
```diff
@@ -119,37 +119,43 @@ class SciTensor(val variables: mutable.HashMap[String, AbstractTensor]) extends
...
    * Masks the current variable array by preserving values
    * greater than num.
    */
   def >(num: Double): SciTensor = variables(varInUse) > num
 
-/**
+  /**
```
Comment (chrismattmann) [inline src/main/scala/org/dia/core/SciTensor.scala]: orthogonal change please remove

File: src/main/scala/org/dia/core/SciTensor.scala [hunk 5]
```diff
@@ -119,37 +119,43 @@ class SciTensor(val variables: mutable.HashMap[String, AbstractTensor]) extends
...
    * Masks the current variable array by preserving values
    * not equal to num.
    */
   def !=(num: Double): SciTensor = variables(varInUse) != num
 
-/**
+  /**
```
Comment (chrismattmann) [inline src/main/scala/org/dia/core/SciTensor.scala]: orthogonal change please remove

**Comment by chrismattmann** [comment]
once the orthogonal changes are removed, I am +1 this is a nice contribution @rahulpalamuttam thanks 👍

**Comment by rahulpalamuttam** [comment]
Didn't see those correcting itself.
I put them back.
Is there a reason why the comment lines are like that - or should I just file  new issue/ make a new PR?

**Comment by chrismattmann** [comment]
thanks for addressing them @rahulpalamuttam LGTM 👍 please merge.

**Comment by kwhitehall** [comment]
THanks @rahulpalamuttam 
Can you also address the issue in convertMa2Arrayto1DJavaArray in NetcdfUtils.scala (and anywhere else), where the data is masked while it is ingested from the netcdf files.

**Comment by rahulpalamuttam** [comment]
@kwhitehall
I didn't find it anywhere else.

**Comment by rahulpalamuttam** [comment]
@kwhitehall 
Anything else you see for this PR?
It looks good to me.

**Comment by kwhitehall** [comment]
It looks good, but I haven't tested it @rahulpalamuttam. Sry for the delay. Will get on it.

**Comment by kwhitehall** [comment]
LGTM @rahulpalamuttam. Thank you!

**Pull request merged**